src/client/utils/tracking/trackOfferEvent.ts

```diff
@@ -69,3 +83,62 @@ export const trackOfferEvent = (
     captureSentryError(error)
   }
 }
+
+export const useTrackOfferEvent = () => {
+  const { isoLocale } = useCurrentLocale()
+
+  const { quoteCartId } = useQuoteCartIdFromUrl()
+
+  const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
+
+  const { data: quoteCartQueryData } = useQuoteCartQuery({
+    variables: {
+      id: quoteCartId,
+      locale: isoLocale,
+    },
+  })
+
+  const isReferralCodeUsed =
+    getMonthlyCostDeductionIncentive(quoteCartQueryData) !== undefined
+
+  const selectedBundleVariant = getSelectedBundleVariant(
+    quoteCartQueryData,
+    selectedInsuranceTypes,
+  )
+
+  const [eventQueue, setEventQueue] = useState<EventParameters[]>([])
+
+  const trackOfferCallback = useCallback(
+    ({ eventName, options = {} }: EventParameters) => {
+      trackOfferEvent(
+        eventName,
+        selectedBundleVariant!.bundle,
+        isReferralCodeUsed,
+        {
+          quoteCartId,
+          ...options,
+        },
+      )
+    },
+    [isReferralCodeUsed, quoteCartId, selectedBundleVariant],
+  )
+
+  const trackOfferHandler = useCallback(
+    (eventParams: EventParameters) => {
+      selectedBundleVariant
+        ? trackOfferCallback(eventParams)
+        : setEventQueue(eventQueue.concat([eventParams]))
+    },
+    [selectedBundleVariant, eventQueue, trackOfferCallback],
+  )
+
+  //cleanup the queue
+  useEffect(() => {
+    if (selectedBundleVariant && eventQueue.length) {
+      eventQueue.map(trackOfferCallback)
+      setEventQueue([])
+    }
```

Comment:
I would avoid map here since we don't care about the return. 

```suggestion
    if (selectedBundleVariant) {
      const event = eventQueue[0]
      if (event) {
        trackOfferCallback(event, selectedBundleVariant)
        setEventQueue((prevQueue) => [...prevQueue].splice(1))
      }
    }
```

Reply:
Right, this was my first thought as well, but figured I'd go with a quick map rather than an effect for each call. It is cleaner like this though, since all events will go through the queue and the callback is only called in one place

Reply:
Right, this was my first thought as well, but figured I'd go with a quick map rather than an effect for each call. It is cleaner like this though, since all events will go through the queue and the callback is only called in one place

Reply:
Yes exactly, better to keep each function doing one thing :)